34-DIPG_data_cleaning.Rmd

```diff
@@ -0,0 +1,287 @@
+
+**Citations:**
+
+> Paugh BS, Broniscer A, Qu C, et al. [Genome-wide analyses identify recurrent 
+amplifications of receptor tyrosine kinases and cell-cycle regulatory genes in 
+diffuse intrinsic pontine glioma.](https://dx.doi.org/10.1200/JCO.2011.35.5677) 
+_J Clin Oncol._ 2011;29(30):3999-4006.
+
+> Buczkowicz P, Hoeman C, Rakopoulos P, et al. [Genomic analysis of diffuse 
+intrinsic pontine gliomas identifies three molecular subgroups and recurrent 
+activating _ACVR1_ mutations.](https://dx.doi.org/10.1038/ng.2936) _Nat Genet._ 
+2014;46(5):451-6. 
+
+## Set up
+
+```{r}
+# magrittr pipe
+`%>%` <- dplyr::`%>%`
+# we need the function that aggregates duplicate gene identifiers to the
+# mean value
+source(file.path("util", "test_LV_differences.R"))
+```
+
+#### Directory setup
+
+```{r}
+# directory that holds the gene expression files
+exprs.dir <- file.path("data", "expression_data")
+# directory that holds the sample metadata
+sample.info.dir <- file.path("data", "sample_info")
+```
+
+## Read in and clean data
+
+### GSE50021
+
+We have the series matrix for `GSE50021`, which contains both the expression
+values and the metadata
+
+```{r}
+series.mat.file <- file.path(exprs.dir, "GSE50021_series_matrix.txt")
+
+# expression matrix -- everything but the comment lines that begin with !
+ma.data <- 
+  readr::read_delim(series.mat.file, 
+                    delim = "\t", 
+                    comment = "!",
+                    col_names = TRUE, 
+                    skip = 1)
+```
+
+#### Gene identifier conversion
+
+```{r}
+# The GPL information from GEO, which was made public on Jul 18, 2011 and 
+# last updated on Jan 18, 2013
+gpl.info.df <- readr::read_tsv(file.path(exprs.dir, "GPL13938-11302.txt"),
+                               comment = "#")
+```
+
+```{r}
+annot.gse50021.df <- gpl.info.df %>%
+  # from the GEO information, grab just the probe identifier and the gene
+  # symbol columns
+  dplyr::select(c(ID, Symbol)) %>%
+  # only ILMN IDs (probes) in both
+  dplyr::inner_join(ma.data, by = c("ID"  = "ID_REF")) %>%
+  # collapsing duplicate symbols later will require the symbols to be in the
+  # first column, called "Gene", with no additional columns
+  dplyr::mutate(Gene = Symbol) %>%
+  dplyr::select(-ID, -Symbol) %>%
+  dplyr::select(Gene, dplyr::everything())
+```
+
+Collapse duplicate symbols and write to file.
+
+```{r}
+# summarize to mean
+annot.mean.df <- PrepExpressionDF(annot.gse50021.df)
+readr::write_tsv(annot.mean.df, file.path(exprs.dir, "GSE50021_mean_agg.pcl"))
+```
+
+#### Sample metadata
+
+As mentioned above, metadata is also extracted from the series matrix file.
+We do this for a single line at a time that we've picked based on the relevance
+to any downstream analysis we might do (contact information, for example, 
+does not help us in this context).
+
+```{r}
+# get more information from the series matrix file
+# sample accession e.g., GSMXXXXX
+conn <- file(series.mat.file)
```

Comment:
Looks like a bunch of file connections opening and closing with the only difference being the `skip` parameter. Is it worth making a function?

Reply:
Good call, thanks!